python/benchmark/test_gen_data.py

```diff
@@ -148,3 +153,60 @@ def test_make_regression(dtype: str, low_rank: bool) -> None:
 
         # Test that y ~= np.dot(X, c) + bias + N(0, 1.0).
         assert_almost_equal(np.std(y - np.dot(X, c)), 1.0, decimal=1)
+
+
+@pytest.mark.parametrize("dtype", ["float32", "float64"])
+@pytest.mark.parametrize("num_rows", [2000, 2001])  # test uneven samples per cluster
+@pytest.mark.parametrize("n_informative, n_repeated", [(31, 0), (28, 3)])
+def test_make_classification(
+    dtype: str, num_rows: int, n_informative: int, n_repeated: int
+) -> None:
+    input_args = [
+        "--num_rows",
+        str(num_rows),
+        "--num_cols",
+        "31",
+        "--dtype",
+        dtype,
+        "--output_dir",
+        "temp",
+        "--output_num_files",
+        "3",
+        "--n_informative",
+        str(n_informative),
+        "--n_redundant",
+        "0",
+        "--n_repeated",
+        str(n_repeated),
+        "--hypercube",
+        "True",
+        "--scale",
+        "0.5",
+        "--flip_y",
+        "0",
+        "--random_state",
```

Comment:
Is there a way to compare with sklearn make_classification to ensure the set of parameters work as expected?

Seems too much work, can move that in the future.

Reply:
Yeah, their tests are pretty complex and make use of params like weights which we don't support. I tried to pick pieces from their tests that would involve as much coverage as possible.